internal/functional/lazyoptional/lazy_optional.go

```
@@ -27,6 +27,24 @@ func Empty[T any]() LazyOptional[T] {
 	}
 }
 
+// OfPtr creates a LazyOptional from a pointer.
+// Returns Empty if the pointer is nil, otherwise contains the dereferenced value.
+func OfPtr[T any](ptr *T) LazyOptional[T] {
+	if ptr == nil {
+		return Empty[T]()
+	}
+	return Some[T](*ptr)
```

golangci-lint (2.5.0)

[error] return with no blank line before

(nlreturn)